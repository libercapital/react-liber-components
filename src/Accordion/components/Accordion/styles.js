import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
