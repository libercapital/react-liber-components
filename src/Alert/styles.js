import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
