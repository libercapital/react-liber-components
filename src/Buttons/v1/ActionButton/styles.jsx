import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
