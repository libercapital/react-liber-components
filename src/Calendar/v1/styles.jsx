import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../../Themes';

import { CardContent } from '../../Card';
import { FaAngleLeft, FaAngleRight } from '../../Icons';
import { Button } from '../../Buttons';
import { media } from '../../Util';

export const Dot = styled.span`
  width: 6px;
  height: 6px;
  background-color: ${props => handleThemeFromObject(props.theme, 'calendar.dot', '#cecece', props.color)};
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  display: inline-block;
  border-radius: 3px;
  margin-right: 8px;
`;

export const BiggerDot = styled(Dot)`
  width: unset;
  height: 18px;
  width: unset;
  min-width: 18px;
  border-radius: 9px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.biggerDot', '#fff')};
  margin-right: 0px;
  margin-bottom: 5px;
  font-weight: bold;
  padding: 0 5px;
  overflow: hidden;
  box-sizing: border-box;
`;

export const FloatingDots = styled.div`
  position: absolute;
  top: 4px;
  left: ${props => (props.today ? '32px' : '27px')};
  width: 18px;
  display: flex;
  overflow: hidden;
  flex-wrap: wrap;
  & ${Dot} {
    margin-bottom: 2px;
    margin-right: 2px;
  }
  ${media.small`
    left: ${props => (props.today ? '36px' : '31px')};
  `}
  ${media.medium`
    left: ${props => (props.today ? '36px' : '31px')};
  `}
  ${media.large`
    left: ${props => (props.today ? '40px' : '35px')};
  `}
  ${media.extraLarge`
    left: ${props => (props.today ? '40px' : '35px')};
  `}
`;

export const DataContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  font-family: inherit;
  font-size: 11px;
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.dataContainer', '#717171')};
  text-transform: none;
  margin-bottom: 12px;
  width: unset;
  ${media.small`
    font-size: 11px;
  `}
  ${media.medium`
    font-size: 11px;
  `}
  ${media.large`
    width: 65px;
    font-size: 13px;
  `}
  ${media.extraLarge`
    width: 65px;
    font-size: 13px;
  `}
`;

const tooltipPosition = (position) => {
  switch (position) {
    case 'right':
      return css`
        bottom: 30px;
        left: 30px;
      `;
    case 'left':
      return css`
        bottom: 30px;
        right: 30px;
      `;
    case 'middle':
    default:
      return css`
        bottom: 40px;
        left: calc(50% - calc(148px / 2));
      `;
  }
};

export const Tooltip = styled.div`
  position: absolute;
  ${props => tooltipPosition(props.position)}
  font-family: inherit;
  font-size: 12px;
  font-weight: 500;
  width: 148px;
  border-radius: 4px;
  background-color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.background', '#ffffff')};
  box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.2);
  color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.color', '#717171')};
  padding: 8px;
  display: ${props => (props.hide ? 'none!important' : 'none')};
  z-index: 1000;
  box-sizing: border-box;
  & > h6 {
    margin: 0px;
    opacity: 0.75;
    font-family: inherit;
    font-size: 12px;
    font-weight: bold;
    text-align: left;
    color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.color', '#717171')};
    text-transform: capitalize;
    text-align: center;
  }
  & > div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
`;

export const Day = styled.div`
  position: relative;
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.date.today.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.date.today.fontSize', '14px')};
  width: 28px;
  height: 28px;
  border: 0px;
  margin: 0px;
  padding: 0px;
  border-radius: 20px;
  user-select: none;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};

  background: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.background', 'linear-gradient(to bottom, #00ccff, #009dff)')
    : 'transparent'
  )};
  box-shadow:  ${props => (props.today ? '0 2px 6px 0 rgba(0, 0, 0, 0.05)' : null)};
  font-weight: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightToday', '700')
    : handleThemeFromObject(props.theme, 'calendar.date.today.fontWeight', '400')
  )};

  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.today.color', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.default', '#717171');
  }};

  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    background-color: rgba(230, 230, 230, 0.5);
    font-weight: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightToday', '700')
    : handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightHighlight', '500')
  )};
  }
  &:hover > ${Tooltip} {
    display: block;
  }
  &:hover ${FloatingDots} {
    display: ${props => (props.showDot ? null : 'none')};
  }
  ${media.small`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.large`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
`;

export const DayLink = styled.a`
  text-decoration: none;
  position: relative;
  font-family: inherit;
  font-size: 14px;
  width: 28px;
  height: 28px;
  border: 0px;
  margin: 0px;
  padding: 0px;
  border-radius: 20px;
  user-select: none;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};

  background: ${props => (props.today
    ? handleThemeFromObject(
      props.theme,
      'calendar.date.link.background',
      'linear-gradient(to bottom, #00ccff, #009dff)',
    )
    : 'transparent'
  )};
  box-shadow:  ${props => (props.today ? '0 2px 6px 0 rgba(0, 0, 0, 0.05)' : null)};
  font-weight: ${props => (props.today ? 'bold' : '500')};

  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.today', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.link.default', '#717171');
  }};

  display: flex;
  justify-content: center;
  align-items: center;


  &:hover {
    background-color: rgba(230, 230, 230, 0.5);
    color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.today', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.link.default', '#717171');
  }};
  }
  &:hover > ${Tooltip} {
    display: block;
  }
  &:hover ${FloatingDots} {
    display: ${props => (props.showDot ? null : 'none')};
  }
  ${media.small`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.medium`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.large`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
  ${media.extraLarge`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
`;

export const WeekDay = styled.div`
  width: 30px;
  height: 24px;
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSize', '12px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontWeight', '700')};
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.week.color', '#009dff')};
  text-transform: uppercase;

  ${media.small`
    width: 35.1px;
    font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSizeSmall', '14px')};
  `}
  ${media.large`
    width: 54.1px;
    font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSizeLarge', '20px')};
  `}
`;

export const Month = styled.div`
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontSize', '20px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontWeight', '500')};
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.month,color', '#717171')};
  text-transform: uppercase;
  width: unset;
  flex-grow: 1;
  ${media.large`
    width: 260px;
    flex-grow: 0;
  `}
`;

export const Week = styled(Month)`
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontSize', '20px')};
  width: unset;
  flex-grow: 1;

  ${media.large`
    width: 250px;
    flex-grow: 0;
  `}
`;

export const Row = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  min-height: 36px;
  max-height: 36px;
  ${media.small`
    min-height: 40px;
    max-height: 40px;
  `}
  ${media.medium`
    min-height: 40px;
    max-height: 40px;
  `}
  ${media.large`
    min-height: 43px;
    max-height: 43px;
  `}
  ${media.extraLarge`
    min-height: 43px;
    max-height: 43px;
  `}
`;

export const FilterRow = styled(Row)`
  justify-content: flex-start;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.filterRow', '#717171')} !important;
  font-size: 14px;
  font-weight: 500;
  align-items: flex-start;
  min-height: unset;
  max-height: unset;
  ${media.large`
    align-items: center;
  `}
  ${media.extraLarge`
    align-items: center;
  `}
`;

export const FiltersBox = styled(FilterRow)`
  flex-wrap: wrap;
  font-size: 10px;
  & > *:not(:last-child) {
    margin-bottom: 10px;
  }
  ${media.large`
    font-size: 14px;
    flex-wrap: nowrap;
    & > * {
      margin-bottom: 0px !important;
    }
  `}
  ${media.extraLarge`
    font-size: 14px;
    flex-wrap: nowrap;
    & > * {
      margin-bottom: 0px !important;
    }
  `}
`;

export const FilterLabel = styled.div`
  font-family: inherit;
  font-size: 14px;
  font-weight: 900;
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.filterLabel', '#717171')};
  margin-right: 20px;
`;

export const Cell = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const CalendarBody = styled(CardContent)`
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${props => (props.week ? '0px' : '10px')};
`;

export const Left = styled(FaAngleLeft)`
  width: 24px;
  height: 24px;
  fill: ${props => handleThemeFromObject(props.theme, 'calendar.left', '#009dff')};
`;

export const Right = styled(FaAngleRight)`
  width: 24px;
  height: 24px;
  fill: ${props => handleThemeFromObject(props.theme, 'calendar.right', '#009dff')};
`;

export const NavButton = styled(Button)`
  background-color: transparent;
  border-radius: 50%;
  padding: 0px;
  width: 40px;
  height: 40px;
  &:hover {
    background-color: ${props => (props.disabled ? null : 'rgba(230, 230, 230, 0.5)')};
  }
  margin: 0px;
  & > svg{
    fill: ${props => (props.disabled ? 'rgba(230, 230, 230, 0.5)' : null)};
  }
  ${media.small`
    margin: 0 15px;
  `}
  ${media.medium`
    margin: 0 15px;
  `}
  ${media.large`
    margin: 0 10px;
  `}
  ${media.extraLarge`
    margin: 0 10px;
  `}
`;

export const WeekBody = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: space-evenly;
  width: 100%;
`;

export const WeekDayValue = styled.div`
  display: inline-flex;
  flex-direction: column;
  font-family: inherit;
  font-size: 18px;
  width: 47px;
  font-weight: 500;
  margin-bottom: 12px;
  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.today', '#009dff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.default', '#717171');
  }};
  align-items: center;
  text-transform: uppercase;
  ${media.large`
    font-size: 20px;
  `}
  ${media.extraLarge`
    font-size: 20px;
  `}
`;

export const WeekCell = styled.div`
  flex-basis: 20%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  min-height: 300px;
  border-image:
    linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0)) 1 100%;
  &:last-child {
    border: 0px;
  }

  padding-top: 5px;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};
  border-top: 0px;
  border-left: 0px;
  &:hover {
    background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0));
  }
  &:hover > ${WeekDayValue} {
    font-weight: 900;
  }
  &:hover ${DataContainer} {
    font-weight: 900;
  }
`;

export const WeekCellLink = styled.a`
  text-decoration: none;
  flex-basis: 20%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  min-height: 300px;
  border-image:
    linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0)) 1 100%;
  &:last-child {
    border: 0px;
  }

  padding-top: 5px;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};
  border-top: 0px;
  border-left: 0px;
  &:hover {
    background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0));
  }
  &:hover > ${WeekDayValue} {
    font-weight: 900;
  }
  &:hover ${DataContainer} {
    font-weight: 900;
  }
`;

export const WeekNav = styled(Row)`
  padding: 15px 0px;
  border-bottom: 2px solid rgba(230, 230, 230, 0.3);
`;
