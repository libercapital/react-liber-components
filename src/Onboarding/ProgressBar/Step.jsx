import React from 'react';
import PropTypes from 'prop-types';
import {
  StepContainer, StepIcon, StepCheck, StepConnect, StepLabel,
} from './styles';

const renderIcon = (number, active, complete, onClick, clickable) => {
  if (complete) {
    return (
      <StepIcon complete={complete} onClick={clickable ? onClick : undefined} clickable={clickable}>
        <StepCheck />
      </StepIcon>
    );
  }
  return (
    <StepIcon active={active} complete={complete} onClick={clickable ? onClick : undefined} clickable={clickable}>
      {number}
    </StepIcon>
  );
};

const renderConnect = (connect, active, complete) => {
  if (connect) {
    return (<StepConnect active={active} complete={complete} />);
  }
  return null;
};

const renderLabel = (label) => {
  if (label) {
    return <StepLabel>{label}</StepLabel>;
  }
  return null;
};

const Step = ({
  active,
  clickable,
  connect,
  complete,
  number,
  label,
  onClick,
}) => (
  <StepContainer active={active} complete={complete}>
    { renderIcon(number, active, complete, onClick, clickable) }
    { renderConnect(connect, active, complete) }
    { renderLabel(label) }
  </StepContainer>
);

Step.propTypes = {
  active: PropTypes.bool,
  clickable: PropTypes.bool,
  connect: PropTypes.bool,
  complete: PropTypes.bool,
  number: PropTypes.number.isRequired,
  label: PropTypes.string,
  onClick: PropTypes.func,
};

Step.defaultProps = {
  active: false,
  clickable: false,
  connect: false,
  complete: false,
  label: '',
  onClick: () => {},
};

export default Step;
