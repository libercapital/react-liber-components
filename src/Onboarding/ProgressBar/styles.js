import styled, { css } from 'styled-components';
import { FaCheck } from '../../Icons';
import { handleThemeFromObject } from '../../Themes';

const background = '#fafafa';
const changeWidth = '992px';

const stepActive = '#009dff';
const stepComplete = '#4cbaff';
const stepInactive = '#a4a4a4';

const stepGradient = (complete, active) => `linear-gradient(to right, ${complete}, ${active})`;

const setConnectColor = (theme, active, complete) => {
  if (active) {
    return stepGradient(
      handleThemeFromObject(theme, 'progressBar.step.complete', stepComplete),
      handleThemeFromObject(theme, 'progressBar.step.active', stepActive),
    );
  }
  if (complete) {
    return handleThemeFromObject(theme, 'progressBar.step.complete', stepComplete);
  }
  return handleThemeFromObject(theme, 'progressBar.step.inactive', stepInactive);
};

const setStepColor = (theme, active, complete) => {
  if (active) {
    return handleThemeFromObject(theme, 'progressBar.step.active', stepActive);
  }
  if (complete) {
    return handleThemeFromObject(theme, 'progressBar.step.complete', stepComplete);
  }
  return handleThemeFromObject(theme, 'progressBar.step.inactive', stepInactive);
};

export const Progress = styled.div`
  background-color: ${props =>
    handleThemeFromObject(props.theme, 'progressBar.background', background)};
  padding-top: 22px;
  padding-bottom: 22px;
`;

const Label = styled.h4`
  font-size: 22px;
  font-weight: bold;
  color: ${props => handleThemeFromObject(props.theme, 'progressBar.label', '#4e4e4e')};
  margin: 0;
`;

export const LabelUpper = styled(Label)`
  @media (max-width: ${changeWidth}) {
    display: none;
  }
`;

export const LabelLower = styled(Label)`
  margin-top: 20px;
  display: none;

  @media (max-width: ${changeWidth}) {
    display: block;
  }
`;

export const Steps = styled.div`
  display: flex;
  flex-direction: row;

  @media (min-width: ${changeWidth}) {
    margin-top: 30px;
  }
`;

export const StepContainer = styled.div`
  position: relative;
  display: inline-block;
  color: ${props => setStepColor(props.theme, props.active, props.complete)};
  width: auto;
  flex: 1;
`;

const StepCell = styled.div`
  position: relative;
  display: block;
  z-index: 2;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  text-align: center;
  margin: 0 auto;
  overflow: hidden;
  box-sizing: initial !important;
  user-select: none;

  ${props =>
    props.clickable
      ? css`
          transition: transform 0.2s ease;
          cursor: pointer;

          &:hover {
            transform: scale(1.1);
          }
        `
      : null}
`;

export const StepIcon = styled(StepCell)`
  border: solid 3px ${props => setStepColor(props.theme, props.active, props.complete)};
  background-color: ${props =>
    props.complete
      ? handleThemeFromObject(props.theme, 'progressBar.step.complete', stepComplete)
      : handleThemeFromObject(props.theme, 'progressBar.background', background)};
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const StepCheck = styled(FaCheck)`
  width: 15px;
  height: 15px;
  fill: white;
`;

export const StepConnect = styled.div`
  position: absolute;
  z-index: 1;
  display: block;
  background: ${props => setConnectColor(props.theme, props.active, props.complete)};
  height: 4px;
  width: calc(100% - 25px);
  top: 13px;
  left: 13px;
  margin-left: -50%;
`;

export const StepLabel = styled.div`
  margin-top: 10px;
  text-align: center;

  @media (max-width: ${changeWidth}) {
    display: none;
  }
`;
