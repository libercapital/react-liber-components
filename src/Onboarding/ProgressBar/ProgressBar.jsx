import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Container } from '../../GenericStyles';
import Step from './Step';
import {
  LabelUpper, LabelLower, Progress, Steps,
} from './styles';

const generateSteps = (steps, completed, current, onClick) => {
  if (typeof steps === 'object') {
    return Object.keys(steps).map((label, index) => {
      const key = _.uniqueId(label);
      const num = index + 1;
      return (
        <Step
          key={key}
          active={current === num}
          connect={num > 1}
          complete={completed ? num <= completed : current > num}
          label={steps[label]}
          number={num}
          clickable={completed ? num <= completed + 1 : false}
          onClick={() => (typeof onClick === 'function' ? onClick(steps[label], num) : {})}
        />
      );
    });
  }

  return steps.map((label, index) => {
    const key = _.uniqueId(label);
    const num = index + 1;
    return (
      <Step
        key={key}
        active={current === num}
        connect={num > 1}
        complete={completed ? num <= completed : current > num}
        label={label}
        number={num}
        clickable={completed ? num <= completed + 1 : Boolean(onClick)}
        onClick={() => onClick(steps[label], num)}
      />
    );
  });
};

const getStepNum = (steps, current) => {
  if (typeof current === 'string') {
    if (Array.isArray(steps)) {
      return steps.findIndex(step => step === current) + 1;
    }
    return Object.keys(steps).findIndex(step => step === current) + 1;
  }
  return current;
};

const ProgressBar = ({
  steps, completed, current, label, onClick,
}) => {
  const stepNum = getStepNum(steps, current);
  return (
    <Progress>
      <Container>
        <LabelUpper>Cadastro | {label || `Etapa ${stepNum}`}</LabelUpper>
        <Steps>
          { generateSteps(steps, completed, stepNum, onClick) }
        </Steps>
        <LabelLower>Cadastro | {label || `Etapa ${stepNum}`}</LabelLower>
      </Container>
    </Progress>
  );
};

ProgressBar.propTypes = {
  steps: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.objectOf(PropTypes.string),
  ]).isRequired,
  completed: PropTypes.number,
  current: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  label: PropTypes.string,
  onClick: PropTypes.func,
};

ProgressBar.defaultProps = {
  completed: undefined,
  label: '',
  onClick: undefined,
};

export default ProgressBar;
