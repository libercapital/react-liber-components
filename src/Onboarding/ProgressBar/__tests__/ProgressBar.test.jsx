import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

import ProgressBar from '../ProgressBar';
import Step from '../Step';
import { LabelUpper } from '../styles';

describe('Test for ProgressBar', () => {
  configure({ adapter: new Adapter() });
  const props = {
    steps: ['A', 'B', 'C'],
    current: 2,
  };

  it('should render with props', () => {
    const wrapper = shallow(<ProgressBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with custom label', () => {
    const wrapper = shallow(<ProgressBar {...props} label="ABC" />);
    expect(wrapper.find(LabelUpper).props().children[1]).toEqual('ABC');
  });

  it('should render with onClick function when completed', () => {
    const onClick = sinon.spy();
    const wrapper = shallow(<ProgressBar {...props} completed={2} onClick={onClick} />);
    const steps = wrapper.find(Step);
    steps.at(0).simulate('click');
    expect(onClick.calledWith('A', 1)).toBeTruthy();
    expect(steps.get(0).props.clickable).toBeTruthy();
    steps.at(1).simulate('click');
    expect(onClick.calledWith('B', 2)).toBeTruthy();
    expect(steps.get(1).props.clickable).toBeTruthy();
    steps.at(2).simulate('click');
    expect(onClick.calledWith('C', 3)).toBeTruthy();
    expect(steps.get(2).props.clickable).toBeTruthy();
  });

  it('should render with onClick function and completed', () => {
    const onClick = sinon.spy();
    const wrapper = shallow(<ProgressBar {...props} completed={1} onClick={onClick} />);
    const steps = wrapper.find(Step);
    expect(steps.get(0).props.clickable).toBeTruthy();
    expect(steps.get(0).props.complete).toBeTruthy();
    expect(steps.get(1).props.clickable).toBeTruthy();
    expect(steps.get(1).props.complete).toBeFalsy();
    expect(steps.get(2).props.clickable).toBeFalsy();
    expect(steps.get(2).props.complete).toBeFalsy();
  });

  it('should render steps with correct props when array', () => {
    const wrapper = shallow(<ProgressBar {...props} />);
    const steps = wrapper.find(Step);
    expect(steps.length).toEqual(3);
    expect(steps.get(0).props.active).toBeFalsy();
    expect(steps.get(0).props.complete).toBeTruthy();
    expect(steps.get(1).props.active).toBeTruthy();
    expect(steps.get(1).props.complete).toBeFalsy();
    expect(steps.get(2).props.active).toBeFalsy();
    expect(steps.get(2).props.complete).toBeFalsy();
  });

  it('should render steps with correct props when object', () => {
    const objProps = {
      steps: { a: 'A', b: 'B', c: 'C' },
      current: 2,
    };
    const wrapper = shallow(<ProgressBar {...objProps} />);
    const steps = wrapper.find(Step);
    expect(steps.length).toEqual(3);
    expect(steps.get(0).props.active).toBeFalsy();
    expect(steps.get(0).props.complete).toBeTruthy();
    expect(steps.get(1).props.active).toBeTruthy();
    expect(steps.get(1).props.complete).toBeFalsy();
    expect(steps.get(2).props.active).toBeFalsy();
    expect(steps.get(2).props.complete).toBeFalsy();
  });

  it('should handle current when array and current as string', () => {
    const wrapper = shallow(<ProgressBar {...props} current="B" />);
    const steps = wrapper.find(Step);
    expect(steps.get(1).props.active).toBeTruthy();
    expect(steps.get(1).props.complete).toBeFalsy();
  });

  it('should handle when object and current as string', () => {
    const objProps = {
      steps: { a: 'A', b: 'B', c: 'C' },
      current: 'b',
    };
    const wrapper = shallow(<ProgressBar {...objProps} />);
    const steps = wrapper.find(Step);
    expect(steps.get(1).props.active).toBeTruthy();
    expect(steps.get(1).props.complete).toBeFalsy();
  });

  it('should render steps with correct connects count', () => {
    const wrapper = shallow(<ProgressBar {...props} />);
    const steps = wrapper.find(Step);
    const connects = steps.filterWhere(step => step.props().connect);
    expect(steps.length).toEqual(3);
    expect(connects.length).toEqual(2);
  });

  it('should render connects with correct props', () => {
    const wrapper = shallow(<ProgressBar {...props} />);
    const steps = wrapper.find(Step);
    const connects = steps.filterWhere(step => step.props().connect);
    expect(connects.get(0).props.active).toBeTruthy();
    expect(connects.get(0).props.complete).toBeFalsy();
    expect(connects.get(1).props.active).toBeFalsy();
    expect(connects.get(1).props.complete).toBeFalsy();
  });
});
