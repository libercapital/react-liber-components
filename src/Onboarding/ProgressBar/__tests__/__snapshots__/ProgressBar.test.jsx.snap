// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ProgressBar should render with props 1`] = `
<styles__Progress>
  <GenericStyles__Container>
    <styles__LabelUpper>
      Cadastro | 
      Etapa 2
    </styles__LabelUpper>
    <styles__Steps>
      <Step
        active={false}
        clickable={false}
        complete={true}
        connect={false}
        key="01"
        label="A"
        number={1}
        onClick={[Function]}
      />
      <Step
        active={true}
        clickable={false}
        complete={false}
        connect={true}
        key="12"
        label="B"
        number={2}
        onClick={[Function]}
      />
      <Step
        active={false}
        clickable={false}
        complete={false}
        connect={true}
        key="23"
        label="C"
        number={3}
        onClick={[Function]}
      />
    </styles__Steps>
    <styles__LabelLower>
      Cadastro | 
      Etapa 2
    </styles__LabelLower>
  </GenericStyles__Container>
</styles__Progress>
`;
